### PR TITLE
Bump Gradle to 8.0

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
I too started working on this before this PR:
https://github.com/apache/iceberg/pull/6825

But I will close mine. As I couldn't finish all the errors.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 # checksum was taken from https://gradle.org/release-checksums
-distributionSha256Sum=7ba68c54029790ab444b39d7e293d3236b2632631fb5f2e012bb28b4ff669e4b
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+distributionSha256Sum=4159b938ec734a8388ce03f52aa8f3c7ed0d31f5438622545de4f83a89b79788
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```
Comment (ajantha-bhat) [inline gradle/wrapper/gradle-wrapper.properties:5]: I think we also need to update here. 
https://github.com/apache/iceberg/blob/ef7e20e12eae2638015ff91b3356eb2f32f601cc/gradlew#L86

**Review by ajantha-bhat — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM with one question

File: build.gradle [hunk 2]
```diff
@@ -219,7 +226,7 @@ project(':iceberg-bundled-guava') {
   }
 
   shadowJar {
-    classifier null
+    archiveClassifier.set(null)
```
Comment (nastra) [inline build.gradle:229]: nit: should `archiveClassifier = 'empty'` be changed to `archiveClassifier.set('empty')` in L251?

Comment (nastra) [reply build.gradle:229]: same for https://github.com/apache/iceberg/blob/2d5a7091c16c167428e0cd6d932d8b59b4677b0f/deploy.gradle#L42

File: build.gradle [hunk 3]
```diff
@@ -258,6 +265,16 @@ project(':iceberg-api') {
 
   tasks.processTestResources.dependsOn rootProject.tasks.buildInfo
 
+  // Workaround to prevent:
+  // Task ':iceberg-api:processTestResources' uses this output of task
+  // ':spotlessInternalRegisterDependencies' without declaring an explicit or implicit dependency.
```
Comment (nastra) [inline build.gradle:270]: do you think this dependency exists because of `tasks.processTestResources.dependsOn rootProject.tasks.buildInfo`?

**Comment by snazy** [comment]
Rebased to resolve conflict

File: build.gradle (hunk 3 above)
Comment (snazy) [reply build.gradle:269]: Maybe. Cross-project task dependencies are never a good idea.

File: build.gradle (hunk 2 above)
Comment (snazy) [reply build.gradle:229]: updated those

**Comment by ajantha-bhat** [comment]
@Fokko: Can you please take a look at this?

**Review by Fokko — APPROVED** [review]
@snazy, thanks for taking the time to bump to Gradle 8.0. Thanks @nastra and @ajantha-bhat for the review and for pinging me. I noticed this PR but didn't get to it yet.

**Pull request merged**